nix/shared.nix

```
@@ -444,7 +444,9 @@ let
       '';
     };
 
-  toShell = drv: drv.env;
+  toShell = drv: drv.env.overrideAttrs (base: {
+    buildInputs = (base.buildInputs or []) ++ [ pkgs.cabal-install ];
```

```suggestion
    nativeBuildInputs = (base.nativeBuildInputs or []) ++ [ pkgs.cabal-install ];
```

This is a minor thing since the `env` derivation isn't actually built, but `nativeBuildInputs` are intended for things that are required only at build time, whereas `buildInputs` is for dependencies that are used at runtime.  The Nixpkgs manual has a longer explanation of this here:

https://nixos.org/manual/nixpkgs/stable/#variables-specifying-dependencies